parsers/US_NEISO.py

```diff
@@ -138,8 +147,98 @@ def fetch_production(zone_key='US-NEISO', session=None, target_datetime=None, lo
     return production_mix
 
 
+def fetch_exchange(zone_key1, zone_key2, session=None, target_datetime=None, logger=None):
+    """Requests the last known power exchange (in MW) between two zones
+
+    Arguments:
+    zone_key1, zone_key2: specifies which exchange to get
+    session (optional): request session passed in order to re-use an existing session
+    target_datetime: the datetime for which we want production data. If not provided, we should
+      default it to now. The provided target_datetime is timezone-aware in UTC.
+    logger: an instance of a `logging.Logger`; all raised exceptions are also logged automatically
+
+    Return:
+    A list of dictionaries in the form:
+    [{
+      'sortedZoneKeys': 'CA-QC->US-NEISO',
+      'datetime': '2017-01-01T00:00:00Z',
+      'netFlow': 0.0,
+      'source': 'mysource.com'
+    }]
+    """
+    sorted_zone_keys = '->'.join(sorted([zone_key1, zone_key2]))
+
+    # For directions, note that ISO-NE always reports its import as negative
+
+    if sorted_zone_keys == 'CA-NB->US-NEISO':
+        # CA-NB->US-NEISO means import to NEISO should be positive
+        multiplier = -1
+
+        postdata = {
+            '_nstmp_zone0': '4010'  # ".I.SALBRYNB345 1"
+        }
+
+    elif sorted_zone_keys == "CA-QC->US-NEISO":
+        # CA-QC->US-NEISO means import to NEISO should be positive
+        multiplier = -1
+
+        postdata = {
+            '_nstmp_zone0': '4012',  # ".I.HQ_P1_P2345 5"
+            '_nstmp_zone1': '4013'  # ".I.HQHIGATE120 2"
+        }
+
+    elif sorted_zone_keys == 'US-NEISO->US-NY':
+        # US-NEISO->US-NY means import to NEISO should be negative
+        multiplier = 1
+
+        postdata = {
+            '_nstmp_zone0': '4014',  # ".I.SHOREHAM138 99"
+            '_nstmp_zone1': '4017',  # ".I.NRTHPORT138 5"
+            '_nstmp_zone2': '4011'  # ".I.ROSETON 345 1"
+        }
+
+    else:
+        raise Exception('Exchange pair not supported: {}'.format(sorted_zone_keys))
```

Comment:
I suggest we raise NotImplementedError here rather than a generic exception.

Reply:
Wouldn't that potentially get confused with NotImplementedError from parsers that don't support target_datetime? What do you think, Oli and Max?

Reply:
Yes it could though most of the parsers already raise NotImplementedError when an invalid exchange is provided. With a good message it's not an issue imo.

Reply:
`NotImplementedError` is fine

Reply:
OK, I will do NotImplementedError in subsequent parsers/edits :+1: